File: sdks/php/src/Plans/Requests/ListPlansRequest.php
<?php

namespace NovaBilling\Plans\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class ListPlansRequest extends JsonSerializableType
{
    /**
     * @var ?bool $isActive Filter by active status
     */
    public ?bool $isActive;

    /**
     * @param array{
     *   isActive?: ?bool,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->isActive = $values['isActive'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/Plans/Requests/ListPlansRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListPlansRequest
{
    /// <summary>
    /// Filter by active status
    /// </summary>
    [JsonIgnore]
    public bool? IsActive { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
